HumanCellAtlas/metadata-schema
# Fixed travis notifications and pre-commit hooks. Fixes #1025

### Release notes

For `.pre-commit-config.yaml`:
- Deleted `human_readable_json` pre-commit hook as it's now called from within the `generate_md_docs` hook.
- Changed `add_docs` hook id to `generate_md_docs` for clarity
- Changed `Adding generated markdown docs to commit` hook name to `Generating markdown documents`

For `add_docs.sh` document:
- Added `python3 src/human_readable_json.py` before git add
- Changed name to `generate_md_docs.sh`
- Moved to `.githooks` folder. It feels a better place for the pre-commit hooks.

For `.travis.yml`:
- Fixed a syntax error where it wasn't notifying updates to the slack channel.

### How to test new changes

- Open branch in your computer
- Make sure you have the pre-commit package installed
- Make breaking errors in schema (e.g., `biomaterial_id` to `biomaterial_false`) 
- Commit. Should commit properly, executing `human_readable_json` on the commit (This tests the changes to the pre-commit hooks).
- Push. Travis build should fail and a notification will be sent to the slack (This tests changes to `travis.yml`)
- Change back to previous state, commit and push. This time the travis build will pass and a notification will be sent to slack.

### Reviews requested

- Need 1 Reviewers to approve because this is a minor scripts update
- Last Reviewer to review/approve, merge changes by 18/06/2019
